### DISCUSSION TITLE
Refreshing a JWT Authentication requires sending an auth packet but no method exists

### DISCUSSION THREAD

**Discussion by sseiber** [mqttjs/MQTT.js#1794] [Q&A] [answered]
I'm using JWT Token authentication sucessfully by setting the mqtt client properties before connecting:
```
properties: {
    authenticationMethod: 'OAUTH2-JWT', // for Azure EventGrid
    authenticationData: <auth-token>
}
```
I need to be able to re-authenticate with the service when the token needs to be refreshed by sending an auth packet:
```
{
    cmd: 'auth',
    reasonCode: 0x19,
    properties: {
        authenticationMethod: 'OAUTH2-JWT',
        authenticationData: <auth-token>
    }
}
```
However, I don't see a way to send a specific packet. I'm following an example that uses MQTTnet to provide this functionality (see SendExtendedAuthenticationExchangeDataAsync):
https://github.com/dotnet/MQTTnet/blob/f82d382f23ec8388ed8a33c48b7ede08e2c18622/Source/MQTTnet/Client/MqttClient.cs#L296

Is there a method or standard pattern in mqtt.js for doing this?

**Comment by robertsLando** [comment, marked as answer]
We don't have a method for that, I have created an issue as reminder: https://github.com/mqttjs/MQTT.js/issues/1795
